lib/spack/spack/test/env.py

```diff
@@ -385,3 +378,40 @@ def test_can_add_specs_to_environment_without_specs_attribute(tmp_path, mock_pac
 
     assert len(env.user_specs) == 1
     assert env.manifest.pristine_yaml_content["spack"]["specs"] == ["a"]
+
+
+@pytest.mark.parametrize("filename", [ev.lockfile_name, "as9582g54.lock", "m3ia54s.json"])
+def test_initialize_from_lockfile(tmp_path, filename):
+    """Test that we can initialize an environment from a valid lockfile"""
+
+    init_file = tmp_path / filename
+    env_dir = tmp_path / "env_dir"
+    init_file.write_text('{ "roots": [] }\n')
+    ev.initialize_environment_dir(env_dir, init_file)
+
+    assert os.path.getsize(env_dir / ev.manifest_name) == len(ev.default_manifest_yaml())
+
+
+def test_cannot_initialize_from_bad_lockfile(tmp_path):
+    """Test that we fail on an incorrectly constructed lockfile"""
+
+    init_file = tmp_path / ev.lockfile_name
+    env_dir = tmp_path / "env_dir"
+
+    init_file.write_text("Not a legal JSON file\n")
+
+    with pytest.raises(ev.SpackEnvironmentError, match="from lockfile"):
+        ev.initialize_environment_dir(env_dir, init_file)
+
+
+@pytest.mark.parametrize("filename", ["random.txt", "random.yaml", ev.manifest_name])
+def test_initialize_from_random_file_as_manifest(tmp_path, filename):
+    """Test that any file not recognized as a lockfile is treated as a manifest"""
+
+    init_file = tmp_path / filename
+    env_dir = tmp_path / "env_dir"
+    init_file.touch()
```

Comment:
Let's use a simple manifest inline, and check later that:
1. We have a `spack.yaml` in `env_dir`
2. We don't have a `spack.lock` in env_dir
3. The content of `spack.yaml` is the same as the one of the init file